apps/aecore/src/aec_pinning_agent.erl

```diff
@@ -20,28 +20,44 @@
     ]).
 
 
+-spec get_pinning_data() -> {ok, map()} | {error, atom()}.
 get_pinning_data() ->
-    {ok, #{epoch  := Epoch,
-           first  := First,
-           last   := _Last,
-           length := _Length
-    }} = aec_chain_hc:epoch_info(),
+    {ok, #{epoch := Epoch,
+           first := First,
+           last  := Last}} = aec_chain_hc:epoch_info(),
+    lager:debug("Get pin data for epoch ~p for leader of block ~p", [Epoch - 1, Last]),
     {ok, BlockHash} = aec_chain_state:get_key_block_hash_at_height(First-1),
     {ok, ChainType} = aec_parent_connector:get_parent_chain_type(),
-    
-    #{epoch             => Epoch-1,
-      height            => First-1,
-      block_hash        => BlockHash,
-      parent_type       => ChainType,
-      parent_network_id => aec_parent_connector:get_network_id()}.
+    PrevEpoch = Epoch - 1,
+    Height = First - 1,
+    case aec_consensus_hc:leader_for_height(Last) of
+      {ok, Leader} ->
+        {ok, #{epoch             => PrevEpoch,
+               height            => Height,
+               block_hash        => BlockHash,
+               parent_payload    => encode_pin_payload(#{epoch => PrevEpoch, height => Height, block_hash => BlockHash}),
+               last_leader       => Leader,
+               parent_type       => ChainType,
+               parent_network_id => aec_parent_connector:get_network_id()}};
+      error ->
+          %% schedule not yet cached
+          {error, last_leader_unknown}
+    end.
 
 
 -spec encode_pin_payload(#{epoch => integer(), height => integer(), block_hash => binary()}) -> binary().
 encode_pin_payload(#{epoch := Epoch, height := Height, block_hash := BlockHash}) ->
-    <<Epoch:64/integer, Height:64/integer, BlockHash/binary>>.
+    EpochHex = list_to_binary(erlang:integer_to_list(Epoch, 16)),
+    HeightHex = list_to_binary(erlang:integer_to_list(Height, 16)),
+    EncBlockHash = aeser_api_encoder:encode(key_block_hash, BlockHash),
+    <<EpochHex/binary, ":", HeightHex/binary, " ", EncBlockHash/binary>>.
```

Comment:
Shouldn't this be a proper serialization? Wasn't there a size limitation (on Bitcoin), then a colon, a space and an API-encoded thing can't be correct?

Reply:
Let's worry about that if it turns out a problem

Reply:
Point made: if parent chain can recognise what is posted, these transaction can be sensored. Possibly a valid point.
In this solution we made it even human readable, so certainly filterable.
Other point made: more streamline if we would serialize in aeser_api_encode for this and have a special type `hc_pin` for this. That is indeed more coherent. 

I made an issue: https://github.com/aeternity/aeternity/issues/4435